src/util/threadpool.rs

```diff
@@ -551,21 +551,33 @@ mod test {
         // the running tasks should be {txn31, txn32}
         assert_eq!(rx.recv_timeout(recv_timeout_duration).unwrap(), group1);
         assert_eq!(rx.recv_timeout(recv_timeout_duration).unwrap(), group1);
-        let mut group2_num = 0;
         let mut group1_num = 0;
-        for _ in 0..4 {
+        let mut group2_num = 0;
+        let mut group3_num = 0;
+        for _ in 0..6 {
             let group = rx.recv_timeout(recv_timeout_duration).unwrap();
-            if group == group1 {
-                group1_num += 1;
-                continue;
+            match group {
+                _ if group == group1 => group1_num += 1,
```

Comment:
I think it's not a good idea to test rely on `sleep` , may be we could use channel instead of `sleep` to fix this bug?